<a href="https://colab.research.google.com/github/TanviPandey10/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*


I have chosen **Lane 2: Refresh / Content Opportunity Scoring**.

I selected this lane because it helps prioritize which content pages should be reviewed first for refresh or improvement. Instead of manually checking thousands of pages, the project will use observable search and engagement signals to rank pages based on their refresh opportunity.

The final output will be a ranked recommendation list that supports content reviewers in making better decisions. This is a decision-support system, not a replacement for human judgment.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Decision

The decision is to identify which content pages should be reviewed first for refresh or improvement.

### Who acts?

The output will be used by content reviewers, SEO specialists, or content managers.

### Action

The reviewer will prioritize the highest-ranked pages for manual review and decide whether they should be refreshed, expanded, rewritten, monitored, or left unchanged.

### Cost of a wrong recommendation

If a low-priority page is recommended, reviewers may waste valuable time reviewing pages that do not need immediate attention.

If an important page with declining performance is missed, valuable traffic and business opportunities may be lost.

### Why ML?

The dataset contains many observable signals such as impressions, clicks, CTR, engagement rate, content age, and freshness. These signals interact in complex ways that are difficult to capture with simple rules. Machine learning can help prioritize pages while still supporting human decision-making.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

print("Dataset Shape:", df.shape)

print("\nContent Type Distribution:")
print(df["content_type"].value_counts())

print("\nTrend Direction Distribution:")
print(df["trend_direction"].value_counts())



Dataset Shape: (30000, 44)

Content Type Distribution:
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

Trend Direction Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## Quick Look at the Data

The starter dataset contains **30,000 rows** and **44 columns**.

Most of the content consists of **keyword articles (27,207 pages)**, followed by **feedly articles (2,096)** and **comparison articles (697)**.

The trend distribution shows that **16,262 pages are declining**, while **5,962 are stable**, **4,388 are growing**, **2,236 are new**, and **1,152 are flat**.

These numbers indicate that the dataset contains enough content and performance variation to build a content opportunity scoring system for prioritizing page reviews.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This work provides decision support for prioritizing content pages for review.

The analysis identifies observed patterns and directional relationships in the available data.

The results do not prove that refreshing a page will improve its performance.

This project does not predict Google's algorithms or establish causal relationships.

The recommendations are intended to support human reviewers rather than replace their judgment.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.